### PR TITLE
board: thingy:52 add mpu9250

### PR DESCRIPTION
**Pull request by guissalustiano**
I'm trying to add the [Nine-axis motion sensor MPU9250](https://infocenter.nordicsemi.com/topic/ug_thingy52/UG/thingy52/hw_description/hw_nine_axis_motion_sensor.html) to the thing:52, he is supplied by the mpu_regulator already described in the device tree.

I follow the ccs example and [this topic](https://devzone.nordicsemi.com/f/nordic-q-a/64653/thingy52-zephyr-rtos-and-sensor-mpu6060-sample-not-working) in nrf forum but when I run I always receive NIO.
image

I'm messing with this for the first time so I need help figuring out the problem

### CODE REVIEW

File: boards/arm/thingy52_nrf52832/board.c [hunk 1]
```diff
@@ -24,7 +24,28 @@ static int pwr_ctrl_init(void)
...
-	k_sleep(K_MSEC(1)); /* Wait for the rail to come up and stabilize */
+	k_sleep(K_MSEC(100)); /* Wait for the rail to come up and stabilize */
+
+	return 0;
+}
+
+static const struct gpio_dt_spec mpu_gpio =
+	GPIO_DT_SPEC_GET(DT_NODELABEL(mpu_pwr), enable_gpios);
+
+static int mpu_pwr_ctrl_init(void)
```
Comment (MaureenHelm) [inline boards/arm/thingy52_nrf52832/board.c:35]: Instead you can update the `status` property of the `mpu_pwr` node in the devicetree

File: boards/arm/thingy52_nrf52832/thingy52_nrf52832.dts [hunk 2]
```diff
@@ -171,6 +171,19 @@
...
+		compatible = "invensense,mpu9250";
+		reg = <0x68>;
+		vin-supply = <&mpu_pwr>;
+		status = "okay";
+		irq-gpios = <&gpio0 06 GPIO_ACTIVE_HIGH>;
+		gyro-sr-div = <10>;
+		gyro-dlpf = <5>;
+		gyro-fs = <250>;
+		accel-fs = <2>;
+		accel-dlpf="5.05";
```
Comment (MaureenHelm) [inline boards/arm/thingy52_nrf52832/thingy52_nrf52832.dts:185]:
```suggestion
		accel-dlpf = "5.05";
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Review by gmarull — CHANGES_REQUESTED** [review]
make sure sensor driver supports regulator (it needs to be done manually now)

**Comment by guissalustiano** [comment]
Oops, I got all messed up with the rebase here, I will open a new PR

**Comment by MaureenHelm** [comment]
> Oops, I got all messed up with the rebase here, I will open a new PR

Please reopen and update this PR instead of creating a new one

**Comment by guissalustiano** [comment]
Apparently, I couldn't anymore. I needed to rewrite all the commits to remove my mess. Sorry.
image

**Comment by martinjaeger** [comment]
You were using your main branch for the PR, which is not recommended.

**Pull request closed without merging**